packages/x-date-pickers/src/DateCalendar/DateCalendar.tsx

```diff
@@ -473,7 +476,7 @@ export const DateCalendar = React.forwardRef(function DateCalendar<TDate>(
         openView={openView}
         currentMonth={calendarState.currentMonth}
         onViewChange={setOpenView}
-        onMonthChange={(newMonth, direction) => handleChangeMonth({ newMonth, direction })}
+        onMonthChange={slideToMonth}
```

Comment:
I now compute the direction inside `useCalendarState` to have a single month change method exposed